packages/web-app-preview/src/App.vue

```diff
@@ -543,6 +524,84 @@ export default defineComponent({
     imageRotateRight() {
       this.currentImageRotation =
         this.currentImageRotation === 270 ? 0 : this.currentImageRotation + 90
+    },
+    isFileTypeImage(file) {
+      return !this.isFileTypeAudio(file) && !this.isFileTypeVideo(file)
+    },
+    isFileTypeAudio(file) {
+      return file.mimeType.toLowerCase().startsWith('audio')
+    },
+
+    isFileTypeVideo(file) {
+      return file.mimeType.toLowerCase().startsWith('video')
+    },
+    addPreviewToCache(file, url) {
+      this.cachedFiles.push({
+        id: file.id,
+        name: file.name,
+        url,
+        ext: file.extension,
+        mimeType: file.mimeType,
+        isVideo: this.isFileTypeVideo(file),
+        isImage: this.isFileTypeImage(file),
+        isAudio: this.isFileTypeAudio(file)
+      })
+    },
+    loadPreview(file) {
+      return loadPreview({
+        resource: file,
+        isPublic: this.isPublicLinkContext,
+        server: configurationManager.serverUrl,
+        userId: this.user.id,
+        token: this.accessToken,
+        dimensions: [this.thumbDimensions, this.thumbDimensions] as [number, number]
+      })
+    },
+    preloadImages() {
+      let toPreloadFiles = []
```

Comment:
Should be an instance variable not a local one, otherwise you will have load images multiple times when you switch between images

Reply:
Good that you chose to track file ids in there, otherwise would have needed invalidation when the current folder changes, for example when clicking a search result 💪